Framework/Geometry/src/Instrument/DetectorInfo.cpp

```diff
@@ -228,6 +228,64 @@ DetectorInfo::signedTwoTheta(const std::pair<size_t, size_t> &index) const {
   return angle;
 }
 
+double DetectorInfo::azimuthal(const size_t index) const {
+  if (isMonitor(index))
+    throw std::logic_error("Azimuthal angle is not defined for monitors");
+
+  const auto samplePos = samplePosition();
+  const auto beamLine = samplePos - sourcePosition();
+
+  if (beamLine.nullVector()) {
+    throw Kernel::Exception::InstrumentDefinitionError(
+        "Source and sample are at same position!");
+  }
+
+  const auto sampleDetVec = position(index) - samplePos;
+  const auto beamLineNormalized = Kernel::normalize(beamLine);
+
+  // generate the vertical axis
+  const auto origHorizontal =
+      m_instrument->getReferenceFrame()->vecPointingHorizontal();
+  const auto vertical = beamLineNormalized.cross_prod(origHorizontal);
```

Comment:
Aren't there cases where this point the in direction opposite to the UP direction defined in the reference frame? Does this mean phi is increasing opposite to the expected direction?